### PR TITLE
Fixed Forced Fraction Bug

### PR DESCRIPTION
**Pull request by michaelkiper**
## Overview
This PR is to resolve an issue with the Forced-Fraction-Simulation test. It was reported that the FPR for the classes were above 0. This should be impossible as the Forced-Fraction-Simulation test set for a particular class should not have any simulated spectra that do not include that class, thus making false positives impossible. 

Screenshot 2026-02-25 at 4 54 24 PM

The simulation is fixed to properly generate the forced fraction test set, so that the FPR for all respective classes will be 0. If it is not, an exception will be raised. 

### New Report:
Screenshot 2026-02-25 at 4 57 27 PM

### CODE REVIEW

File: src/cover_class/reporting/pdf_report.py [hunk 1]
```diff
@@ -168,8 +168,6 @@ def add_section(
     if fractional_simulation_test_dict is not None:
         _metrics_table(contents, report_styles, fractional_simulation_test_dict['TPR'], "Fractional Simulation TPR")
         contents.append(Spacer(1, 0.1 * inch))
-        _metrics_table(contents, report_styles, fractional_simulation_test_dict['FPR'], "Fractional Simulation FPR")
```
Comment (michaelkiper) [inline src/cover_class/reporting/pdf_report.py:171]: Remove FPR report as it should always be 0

File: src/cover_class/reporting/report_config.py [hunk 2]
```diff
@@ -121,19 +122,17 @@ def make_report(self, y_hat: Union[Tensor, NDArray], class_thresholds: Optional[
...
-            self._fractional_simulation_test_dict['FPR'][class_name] = {}
             for f in self.fractional_simulation_test_results:
                 if f.class_id != i:
                     continue
                 f_y_hat_binary = (f.y_hat >= torch.tensor(thresh, device=f.y_hat.device, dtype=f.y_hat.dtype)).to(torch.long)
                 rates = tpr_fpr(f_y_hat_binary, f.y, class_names)[class_name]
                 rname = f'{f.range_low*100} - {f.range_high*100} %'
                 self._fractional_simulation_test_dict['TPR'][class_name][rname] = rates['TPR']
-                self._fractional_simulation_test_dict['FPR'][class_name][rname] = rates['FPR']
+                assert round(rates['FPR'], 5) == 0., f"Unexpected FPR > 0 for {class_name} for the forced fraction simulation - got {rates['FPR']}"
```
Comment (michaelkiper) [inline src/cover_class/reporting/report_config.py:135]: Add in assert that the FPR should always be 0 for the forced fraction simulation since the class should always be present.
